### ISSUE TITLE
Feedback on latest voting page changes

### ISSUE THREAD

**Issue by Shark-93** [top-gg/issues#846]
#### New voting page behavior negative feedback

I have negative feedback of the new implementation of voting, where the vote button is blocked if you have already voted ...

Before the update you would be able to activate the button after the advertisement even if you had already voted; if you tried voting before the due time it would then error out.

Now due to this update, you can unlock the vote button only after the due time runs out so for every vote you have the delay of the advertisement (because after the 12 hours elapse, you have to reload the page; you can't just be ready to vote as you could before, because now the button is blocked).

I'm not sure I explained myself. Basically I prefer to still see the ad and have the vote button available for click, even when I have already voted for the bot: the bot will send me a Discord message when I can vote again and in that second I press Vote.

Now I have to reload the page and wait for the ad. Because at the 12th hour mark even if I'm fully ready to press on the vote I now have to reload the page a second time in order to unlock the vote.

**Comment by KandiPanda** [comment]
Agreed.

Until the change, my vote procedure was to sync my system clock (because otherwise it drifts a bit), open a clock with a seconds display, load the vote pages for the couple bots I vote for a couple minutes ahead of time, let the ads do their thing, and then 12h2s after my previous votes press the vote buttons. (I would do it exactly on, but even with resyncing the clock, that doesn't always work out, so I added a 2 second buffer.)

Now instead of just clicking vote at 2 seconds past, I have to refresh the page, wait for the ad to play (putting me noticeably more than 2 seconds past my previous vote times), repeat for the next vote (likely putting me further still past), and... yeah, the old behavior was better.

(As a side note, if it goes back to the old better behavior, perhaps consider making it 11h59m55s or similar instead of a true 12h, to compensate for things like lag spikes, clock drift, etc? Would be _really_ nice for my vote times to not slowly drift later and later each day.)

**Comment by FlyingndCoding** [comment]
Very much Indeed!

**Comment by Shark-93** [comment]
No replies from Devs yet? I've seen this link in another issue, will submit my feedback there too, please upvote it: https://feedback.top.gg/26

**Comment by TheAlienKnight** [comment]
This seems like unnecessary work for the dev team, and quite frankly it seems like a niche issue. Not to sound rude, but just reload the page? If you are delayed by let's say 10 seconds, 4-5s wait for reload, and 5s for the ad- after **a year** you will be off by 42min. 70 second delay a week give or take.

In my personal opinion, this suggestion/request seems weirdly nitpicky for something relatively simple to do, that already has a solution (reload) that was mentioned above in the original request/feedback post.

**Comment by Shark-93** [comment]
You're not rude at all, but my feedback comes as consequence to a **procedural change** in a well established voting behaviour (so perhaps a Dev could care to explain why that was changed: maybe there's a good technical reason, maybe not, I don't know.. but this also means I that I can't see any benefits from the change either :dizzy_face:).

Your calculation is also incorrect, twofold: you vote twice a day so (it's being assumed) 140s per week and there's 52 weeks in a year, making it 2h total not 94m. Finally we're assuming _zero additional delays,_ which isn't true in real-world use case (so those 2m per week not only force vote-time drifting like @KandiPanda said - more on it below - but exacerbate any daily life delays).

So on top that you now need to add those 2 extra minutes every week (which weren't there 3 weeks ago), literally wasted at looking at a button activate itself. Tell you more, in the old behaviour they even got more "advertisement time" too because now the ad is only played on the second reload, therefore likely closed right after 5s compared to _playing until the 12 hour mark elapses._

So whilst 3 weeks ago one could make sure their voting time was **always** consistent (for eg. at `9:00:00`/`21:00:00` every day, assuming they vote like clockwork which is something I guess a good chunk of users do), now we are forced to the following pattern: `9:00:00`, `21:00:05`, `9:00:10`, `21:00:10` -> your voting time can now **never** be the same again!

All of this wasn't the case 3 weeks ago, the change has made the voting experience worse or rather: it punishes those who vote like clockwork, and for what benefit? And let me tell you regarding being nitpicky - which I totally am eh, no denying it :relieved: - for 3 people that speak up there'll be 30 others that think the same but can't be bothered to speak their mind (or don't even know how to).

**TL;DR** for a voting website where users exist who need to be voting regularly and like clockwork, forcing their voting-time to constantly drift away is a step back and a "punishment" for them.

**Comment by Kiriot22** [comment]
I'm actually fond of the change, since not all bots have the feature to ping you when you can vote again, and the site notifications don't really work, so I end up checking the page every now and then to see if I can vote again.

If you ask me why I can't just set a reminder or something - school hours effectively prevent me from following the 12h time perfectly, so I simply check the vote page during every passing period.

Therefore I think a good compromise would be to keep the "already voted" part BUT make the button enabled.

**Comment by Shark-93** [comment]
Unfortunately I'm afraid that compromise makes even less sense to them, but thank you for trying to come up with something that works for both parties.

Now the problem has gotten even worse at times, because on mobile it'll ask you to fill a recaptcha to cast your vote (it only happened once so far, ergo perhaps I was just unlucky).

At this point it looks like the voting system will not be changed back, ever.
Only a few of us seem to have been bugged by this - somewhat historical? - change.

I say _historical_ because I don't know how long the old system had been in place (where the button is always enabled, even when you've already voted).

**Comment by kortzjwmerkt4ca** [comment]
.























































































.

**Comment by kortzjwmerkt4ca** [comment]
.



















































.

**Comment by kortzjwmerkt4ca** [comment]
.




























.

**Comment by kortzjwmerkt4ca** [comment]
.





















































.

**Comment by kortzjwmerkt4ca** [comment]
@everyone

**Comment by kortzjwmerkt4ca** [comment]
@miramallows

**Comment by Shark-93** [comment]
If a techie staff is reading this, I'm interested to know if there was any technical reason in late 2021 as to why devs decided to play _less advertisements_ to enable the vote button (and thus the advertisement) only when it was actually time to vote (disabling button _and adv_ otherwise).

For example if there were too many calls to the /vote API or whatever, and therefore they had to change the page behaviour to block the vote button from sending numerous vote requests that would result in failed return codes.

Unfortunately the point of the feedback was that the vote button activates only when it's due time to vote, therefore _the advertisement delay_ now increases the voting time little by little every day, forcing a group of users to constantly re-adjust their voting alarms / voting times every few days.

**Comment by Xetera** [comment]
The idea behind the change was to not force users to wait through ads and be greeted with an error when they attempt to vote before the correct time. Yes that results in fewer ads being played, but I think the change aligns with what most people expect when they land on the vote page over the previous behavior which we didn't want users thinking was a dark pattern for ad money or anything like that.

The voting experience as a whole is something we're looking into improving right now which includes the time drifting issue you've brought up. We look at feedback submitted in https://feedback.top.gg if you want to elaborate more on your problem there and have others who share the same experience vote on it. We've been speaking to bot developers to see what areas we can improve and it's good to have feedback from users as well. 

We went about this change based on what we believe is the behavior of majority of users who go on the vote page when it's time to vote, it wasn't targeted to mess with anyone else's voting habits in particular.

**Comment by Shark-93** [comment]
> No replies from Devs yet? I've seen this link in another issue, will submit my feedback there too, please upvote it: [new 2022 feedback entry #129](https://feedback.top.gg/129)

Well I am finally pleased to receive a reply from staff, and it's uncanny your ability to pacify things and gently give perspective to literally anything you say; that is truly a remarkable quality.

I did post in the feedback page back in November but it looks like it wasn't a very popular submission. Now time-drifting of the voting time is basically the whole point of this issue, as @KandiPanda also touched upon before.

Thank you for your words, I will submit a feedback entry that is more on point rather than focusing on the "policy" or page behaviour as I did last time (maybe that's why it got downvoted).
